saleor/core/jwt.py

```diff
@@ -96,13 +95,24 @@ def get_token_from_request(request: WSGIRequest) -> Optional[str]:
 
 def get_user_from_payload(payload: Dict[str, Any]) -> Optional[User]:
     user = User.objects.filter(email=payload["email"], is_active=True).first()
-    if user and user.jwt_token_key == payload["token"]:
-        return user
-    return None
+    user_jwt_token = payload.get("token")
+    if not user_jwt_token or not user:
+        raise InvalidTokenError(
+            "Invalid token. Create new one by using tokenCreate mutation."
+        )
+    if user.jwt_token_key != user_jwt_token:
+        raise ExpiredUserSignatureError(
+            "User requested to expire this token. Create new one by using tokenCreate "
```

Comment:
Isn't it the same as "token expired" error from the API-user point of view? Not sure what is the use case of this exception but I don't know the whole context probably.

Reply:
Unless we don't have any exception for token expiration. But still, I'm wondering if we have to say "User requested to expire this token" in the message.

Reply:
I'm with @maarcingebala here but I'm not sure we should be treating this the same as token expired or as an invalid token. Clients will likely attempt to refresh expired tokens which will fail in this case so it's a wasted API call.